### PR TITLE
[4.0] Ignore failed action log notifications

### PR DESCRIPTION
**Pull request by SharkyKZ**
### Summary of Changes

When sending action log notification emails fails, saving content items fails. This change behavior to ignore failed notifications.
Also suppresses warning message when sending mail is disabled in Global Configuration.

### Testing Instructions

1)

As super admin, edit your account.
In `User Actions Log Options` enable `Email Notifications` option and email to something invalid `me@localhost`.
As a different user, do something loggable, e.g. login or edit your profile.

2) In Global Configuration disable `Send Mail` option. Do something loggable.

### Expected result

1) Works.
2) No messages.

### Actual result

1) >Invalid address (to): me@localhost
2) >The mail function has been disabled by an administrator.

### Documentation Changes Required

IDK.

### CODE REVIEW

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 8b60a8eb83facd887a08d3ce55189164ccaf2e2d<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27548">issues.joomla.org/tracker/joomla-cms/27548</a>.</sub>

File: administrator/components/com_actionlogs/Model/ActionlogModel.php [hunk 1]
```diff
@@ -103,10 +107,16 @@ public function addLog($messages, $messageLanguageKey, $context, $userId = null)
 	 *
 	 * @since   3.9.0
 	 *
-	 * @throws  Exception
+	 * @throws  phpmailerException  if sending mail failed
 	 */
 	protected function sendNotificationEmails($messages, $username, $context)
 	{
+		// Check if send mail option is enabled.
+		if (!Factory::getContainer()->get('config')->get('mailonline'))
```
Comment (mbabker) [inline administrator/components/com_actionlogs/Model/ActionlogModel.php]: The fact this check is necessary shows that the architecture needs to be fixed.

[This line](https://github.com/joomla/joomla-cms/blob/abfd8bac911f6f8ed9453a2f9651ec8db8546bdd/libraries/src/Mail/Mail.php#L163) should be an exception (I'd make a `Joomla\CMS\Mail\Exception\MailDisabledException extends \RuntimeException {}` that way there's a specific exception to be handled), not an `enqueueMessage()` call.

**Comment by jwaisner** [comment]
I have tested this item :white_check_mark: successfully on 8b60a8eb83facd887a08d3ce55189164ccaf2e2d<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27548">issues.joomla.org/tracker/joomla-cms/27548</a>.</sub>

File: administrator/components/com_actionlogs/Model/ActionlogModel.php (hunk 1 above)
Comment (SharkyKZ) [reply administrator/components/com_actionlogs/Model/ActionlogModel.php]: Can you clarify, should we use the same exception in both cases:

1. When Send Mail option in app configuration is disabled
2. When `mail()` function is disabled

?

Or leave second case as `RuntimeException` as it is now?

Comment (mbabker) [reply administrator/components/com_actionlogs/Model/ActionlogModel.php]: Same exception in both cases.  Mail sending is effectively disabled, whether that be through the end user configuration or the platform.

Comment (mbabker) [reply administrator/components/com_actionlogs/Model/ActionlogModel.php]: If you want you can put a reason code into the exception that allows someone to get more details as to why sending is disabled in their error handling code.

```php
namespace Joomla\CMS\Mail\Exception;

final class MailDisabledException extends \RuntimeException
{
    public const REASON_USER_DISABLED = 'user_disabled';
    public const REASON_MAIL_FUNCTION_NOT_AVAILABLE = 'mail_function_not_available';

    /**
     * @var string
     */
    private $reason;

    public function __construct(string $reason, string $message = '', $code = 0, \Throwable $previous = null)
    {
        parent::__construct($message, $code, $previous);

        $this->reason = $reason;
    }

    public function getReason(): string
    {
        return $this->reason;
    }
}
```

The constants just define some standard error codes from the core library, but also allows anyone to use the exception and give their own reason code.

**Comment by SharkyKZ** [comment]
Thanks for tips and tests. Putting this on hold until https://github.com/joomla/joomla-cms/pull/27551 is merged.

**Comment by SharkyKZ** [comment]
PR updated. It can be tested now.

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on e66e9f53919a2045fc33c25424b7e1c22962a107<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27548">issues.joomla.org/tracker/joomla-cms/27548</a>.</sub>

**Comment by jwaisner** [comment]
I have tested this item :white_check_mark: successfully on e66e9f53919a2045fc33c25424b7e1c22962a107<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27548">issues.joomla.org/tracker/joomla-cms/27548</a>.</sub>

**Comment by jwaisner** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27548">issues.joomla.org/tracker/joomla-cms/27548</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**